### DISCUSSION TITLE
AddChannel use case not covered by EnsureChannelFirst (?)

### DISCUSSION THREAD

**Discussion by demcbs** [Project-MONAI/MONAI#5541] [Q&A] [answered]
I have a part of my program where I load in an image manually and want to run it through my SlidingWindowInferer. 

With monai 0.8.1 I used to add a channel dimension with AddChannel to the (3D) image, so the Inferer works. Since upgrading to 1.01 this stopped working, but adding another channel (as a mock batch dimension) made it work again. With the deprecation warning for AddChannel, however, this would stop working in the future and EnsureChannelFirst is not meant to do the same thing.

So I wonder, if there is a case for keeping AddChannel, or is it unintended that SlidingWindowInferer only works with a batched input?

**Comment by KumoLiu** [comment, marked as answer]
Hi @demcbs, if you don't have channel dim, I think you could use `EnsureChannelFirst(channel_dim="no_channel")` to do the same thing with `AddChannel `.
Hope it can help you, thanks!

**Reply by demcbs** [reply]
It seems like a bit of a step backwards to have to use a transform called EnsureChannelFirst to add a batch dimension, but your suggestion works, so I guess I'll just add a comment in the code. Thanks for the quick reply!
